### ISSUE TITLE
0.0.40 regression: `ClassVar["staticmethod[...]"]` assigned from a Callable-typed alias flagged as invalid-assignment

### ISSUE THREAD

**Issue by adamtheturtle** [astral-sh/ty#3569]
### Summary

In ty 0.0.40, a `ClassVar["staticmethod[[str], str]"] = staticmethod(identity)` attribute (where `identity` is annotated `Callable[[str], str]`) on a frozen dataclass whose metaclass declares the same attribute as `Callable[[str], str]` is newly flagged with `invalid-assignment`. ty reports the RHS as `(str, /) -> str` rather than `staticmethod[(str, /), str]`. ty 0.0.39 accepted this; pyright, pyrefly, and mypy all accept it.

### Repro (no external deps)

`repro.py`:
```python
import dataclasses
from collections.abc import Callable
from typing import ClassVar


class LanguageCls(type):
    format_constructor_target: Callable[[str], str]


def _identity(class_name: str, /) -> str:
    return class_name


identity: Callable[[str], str] = _identity


@dataclasses.dataclass(frozen=True, kw_only=True)
class Zig(metaclass=LanguageCls):
    format_constructor_target: ClassVar["staticmethod[[str], str]"] = staticmethod(
        identity,
    )
```

Run:
```bash
uvx ty==0.0.40 check repro.py
```

### Actual result

```
error[invalid-assignment]: Object of type `(str, /) -> str` is not assignable to attribute `format_constructor_target` of type `staticmethod[(str, /), str]`
  --> repro.py:19:5
   |
19 |     format_constructor_target: ClassVar["staticmethod[[str], str]"] = staticmethod(
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^
```

### Expected result

No diagnostic. `staticmethod(identity)` is at runtime a `staticmethod` instance and matches the explicit `staticmethod[[str], str]` annotation. ty appears to unwrap the `staticmethod(...)` call at the assignment site (yielding the plain callable signature) rather than recognising it as a `staticmethod[...]` descriptor.

The sensitivity is to the `Callable`-typed alias: replacing `identity` (typed alias) with `_identity` (plain function) suppresses this particular error.

### Cross-checker results

| Checker | Result |
|---|---|
| `uvx ty==0.0.40 check repro.py` | **ERROR (regression)** |
| `uvx ty==0.0.39 check repro.py` | clean |
| `uvx pyright==1.1.409 repro.py` | clean |
| `uvx pyrefly==1.0.0 check repro.py` | clean |
| `uvx mypy==2.1.0 --strict --config-file=/dev/null repro.py` | clean |

### Notes

#491 is in the same area (Callable ClassVar as bound-method descriptor) but covers `ClassVar[Callable[...]]`, not the explicit `ClassVar["staticmethod[...]"]` annotation used here.

**Comment by charliermarsh** [comment]
I think this one is a regression (but it may have been passing before for the wrong reasons).
